tests/test_sector2fluxnum.py

```diff
@@ -1,13 +1,86 @@
 import os
+from pathlib import Path
+
+import pytest
 
 from subscript.sector2fluxnum import sector2fluxnum
 
+TESTDATA = Path(__file__).absolute().parent / "testdata_sector2fluxnum"
+
+
+def test_main_test(tmp_path, mocker):
+    os.chdir(tmp_path)
+
+    input_ECL_CASE = TESTDATA / "TEST.DATA"
+    input_OUTPUT_FLUX = "OUT_COARSE.FLUX"
+    input_INPUT_DUMPFLUX = TESTDATA / "DUMPFLUX_TEST.DATA"
+    input_RESTART = TESTDATA / "TEST.UNRST"
+
+    mocker.patch(
+        "sys.argv",
+        [
+            "sector2fluxnum",
+            "-i",
+            "2-10",
+            "-j",
+            "2-9",
+            "-k",
+            "1-1",
+            "-r",
+            str(input_RESTART),
+            "--test",
+            str(input_INPUT_DUMPFLUX),
+            str(input_ECL_CASE),
+            str(input_OUTPUT_FLUX),
+            "--ecl_version",
+            "2014.2",
+        ],
+    )
+    sector2fluxnum.main()
+
+    assert Path("USEFLUX_TEST.DATA").exists()
+
 
-def test_main(tmpdir, mocker):
-    TESTDATA = os.path.join(os.path.dirname(__file__), "testdata_sector2fluxnum")
+def test_main_test_fipnum(tmpdir, mocker):
+    """Test the --fipnum command line argument"""
     tmpdir.chdir()
 
-    input_ECL_CASE = os.path.join(TESTDATA, "TEST.DATA")
+    input_ECL_CASE = TESTDATA / "TEST.DATA"
+    input_OUTPUT_FLUX = "OUT_COARSE.FLUX"
+    input_INPUT_DUMPFLUX = TESTDATA / "DUMPFLUX_TEST.DATA"
+    input_RESTART = TESTDATA / "TEST.UNRST"
+
+    mocker.patch(
+        "sys.argv",
+        [
+            "sector2fluxnum",
+            "--fipnum",
+            "4",
+            "-r",
+            str(input_RESTART),
+            "--test",
+            str(input_INPUT_DUMPFLUX),
+            str(input_ECL_CASE),
+            str(input_OUTPUT_FLUX),
+            "--ecl_version",
+            "2014.2",
+        ],
+    )
+    sector2fluxnum.main()
+
+    assert Path("USEFLUX_TEST.DATA").exists()
+
+
+@pytest.mark.skipif(
+    not Path("/prog/res/ecl/grid").exists(),
+    reason="Eclipse must be installed for this test",
+)
+def test_main_notest(tmp_path, mocker):
+    """Test without --test on the command line, requiring
+    Eclipse simulator installed in PATH"""
+    os.chdir(tmp_path)
+
+    input_ECL_CASE = TESTDATA / "TEST.DATA"
     input_OUTPUT_FLUX = "OUT_COARSE.FLUX"
     input_INPUT_DUMPFLUX = os.path.join(TESTDATA, "DUMPFLUX_TEST.DATA")
```

Comment:
Should we use `os.path.join` for all input, for consistency?

Reply:
good catch, changed from os.path.join to pathlib path merging, for consistencyy